TASVideos/Pages/Search/Advanced.cshtml.cs

```diff
@@ -0,0 +1,163 @@
+﻿using System.ComponentModel.DataAnnotations;
+using System.Text.RegularExpressions;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.EntityFrameworkCore;
+using TASVideos.Data;
+using TASVideos.Data.Entity;
+using TASVideos.Data.Entity.Forum;
+
+namespace TASVideos.Pages.Search;
+
+[AllowAnonymous]
+public class AdvancedModel : PageModel
+{
+	public const int PageSize = 10;
+	public const int PageSizeSingle = 50;
+	private readonly ApplicationDbContext _db;
+
+	public AdvancedModel(ApplicationDbContext db)
+	{
+		_db = db;
+	}
+
+	[FromQuery]
+	[StringLength(100, MinimumLength = 2)]
+	[Display(Name = "Search Terms")]
+	public string SearchTerms { get; set; } = "";
+
+	[FromQuery]
+	public int PageNumber { get; set; } = 1;
+
+	public List<PageSearchModel> PageResults { get; set; } = new();
+	public List<TopicSearchModel> TopicResults { get; set; } = new();
+	public List<PostSearchModel> PostResults { get; set; } = new();
+	public List<GameSearchModel> GameResults { get; set; } = new();
+
+	[FromQuery]
+	[Display(Name = "Search Wiki")]
+	public bool PageSearch { get; set; } = false;
+
+	[FromQuery]
+	[Display(Name = "Search Forum Topics")]
+	public bool TopicSearch { get; set; } = false;
+
+	[FromQuery]
+	[Display(Name = "Search Forum Posts")]
+	public bool PostSearch { get; set; } = false;
+
+	[FromQuery]
+	[Display(Name = "Search Games")]
+	public bool GameSearch { get; set; } = true;
+
+	public int DisplayPageSize { get; set; } = PageSize;
+	public bool EnablePrev { get; set; } = false;
+	public bool EnableNext { get; set; } = false;
+
+	public async Task<IActionResult> OnGet()
+	{
+		if (!_db.Database.IsNpgsql())
```

Comment:
What postgres specific functions are you utilizing that need this check?

Reply:
It was copied from the other Search. Since there was no explanation why it was put there I left it there.
I don't know if this is needed.

Reply:
Okay, I removed the check.